client/components/domains/register-domain-step/index.jsx

```diff
@@ -541,10 +539,20 @@ const RegisterDomainStep = React.createClass( {
 				severity = 'info';
 				break;
 			case 'not_registrable':
-				if ( domain.indexOf( '.' ) ) {
-					message = this.translate( 'Sorry but %(domain)s cannot be registered on WordPress.com.', {
-						args: { domain: domain }
-					} );
+				const tldIndex = domain.lastIndexOf( '.' );
+				if ( tldIndex !== -1 ) {
+					message = this.translate(
+						'To use a domain ending with {{strong}}%(tld)s{{/strong}} on your site, ' +
+						'you can register it elsewhere and then add it here. {{a}}Learn more{{/a}}.',
+						{
```

Comment:
That's the copy @ranh provided in https://github.com/Automattic/wp-calypso/issues/6810#issuecomment-233879552. I'm just but a lowly JavaScript programmer - who am I to question His wise words? 😜